File: 0897-increasing-order-search-tree/0897-increasing-order-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    TreeNode dummy = new TreeNode(1);
    TreeNode prev = dummy;
    
    public void makeBSTLinear(TreeNode node) {
        if(node == null) return;
        //Left Recursive Call
        makeBSTLinear(node.left);
        //INORDER ---->
        //Make Current Node null
        node.left = null;
        //Attach node to prev's Right
        prev.right = node;
        //Move Prev ahead
        prev = node;
        //INORDER ---->
        //Right Recursive Call
        makeBSTLinear(node.right);
    }
    
    public TreeNode increasingBST(TreeNode root) {
        makeBSTLinear(root);
        return dummy.right;
    }
}

File: 0897-increasing-order-search-tree/0897-increasing-order-search-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* dummy = new TreeNode(-1);
    TreeNode* prev = dummy;
    
    void makeBSTLinear(TreeNode* node) {
        //Base Case.
        if(node == nullptr) return;
        
        //Left Recursive Call
        makeBSTLinear(node->left);
        //INORDER ------>>>>
        
        //Make Current node's left null;
        node->left = nullptr;
        
        //Initialize prev right to node in Inorder
        prev->right = node;
        
        // Move Previous Ahead
        prev = node;
        //INORDER ------>>>>
        //Right Recursive Call
         makeBSTLinear(node->right);
    }
        
    TreeNode* increasingBST(TreeNode* root) {
        makeBSTLinear(root);
        return dummy->right;
    }
};